cmake/CMakeLists.txt

```diff
@@ -422,15 +422,25 @@ if(WITH_FFMPEG)
   add_definitions(-DLAMMPS_FFMPEG)
 endif()
 
+if(BUILD_SHARED_LIBS)
+  set(CONFIGURE_REQUEST_PIC "--with-pic")
+  set(CMAKE_REQUEST_PIC "-DCMAKE_POSITION_INDEPENDENT_CODE=${CMAKE_POSITION_INDEPENDENT_CODE}")
+else()
+  set(CONFIGURE_REQUEST_PIC "")
+  set(CMAKE_REQUEST_PIC "")
+endif()
+
+
 if(PKG_VORONOI)
-  option(DOWNLOAD_VORO "Download voro++ (instead of using the system's one)" OFF)
+  option(DOWNLOAD_VORO "Download and compile the Voro++ library instead of using an already installed one" OFF)
   if(DOWNLOAD_VORO)
+    message(STATUS "Voro++ download requested - we will build our own")
     include(ExternalProject)
 
     if(BUILD_SHARED_LIBS)
-        set(VORO_BUILD_OPTIONS "CFLAGS=-fPIC")
+      set(VORO_BUILD_OPTIONS CXX="${CMAKE_CXX_COMPILER}" CFLAGS="${CMAKE_POSITION_INDEPENDENT_CODE} ${CMAKE_C_FLAGS}")
```

Comment:
Hmm, I think `${CMAKE_POSITION_INDEPENDENT_CODE}` is just a boolean, I guess we need to use `${CMAKE_SHARED_LIBRARY_CXX_FLAGS}` here.

Reply:
`CFLAGS="${CMAKE_SHARED_LIBRARY_CXX_FLAGS} ${CMAKE_C_FLAGS}"`